tests/test_integration.py

```diff
@@ -254,29 +256,77 @@ def test_integration_feasible_constant_constraint():
     assert actual_entries[2] == pytest.approx(3)
 
 
-# def test_integration_sos1_constraint():
-#     """Test problem with SOS1 constraint"""
-#     # Test with three binary variables where exactly one must be 1
-#     x1 = DecisionVariable.binary(1)
-#     x2 = DecisionVariable.binary(2)
-#     x3 = DecisionVariable.binary(3)
-
-#     instance = Instance.from_components(
-#         decision_variables=[x1, x2, x3],
-#         objective=x1 + 2*x2 + 3*x3,  # Should select x1=1 as it has lowest cost
-#         constraints=[],
-#         sense=Instance.MINIMIZE,
-#     )
-
-#     adapter = OMMXGurobipyAdapter(instance)
-#     model = adapter.solver_input
-#     model.optimize()
-#     state = adapter.decode_to_state(model)
-
-#     actual_entries = state.entries
-#     # Should select x1=1 as it has lowest coefficient in objective
-#     assert actual_entries[1] == pytest.approx(1)
-#     assert actual_entries[2] == pytest.approx(0)
-#     assert actual_entries[3] == pytest.approx(0)
-#     # Sum should be exactly 1 due to SOS1 constraint
-#     assert sum(actual_entries.values()) == pytest.approx(1)
+def test_integration_without_sos1_constraint():
+    """Test problem without SOS1 constraint"""
+    # Objective function: x1 + 2 * x2 + 3 * x3
+    # Constraints:
+    #     x1 + x2 + x3 <= 4
+    #     x1, x2, x3: binary
+    # Optimal solution: x1 = 1, x2 = 1, x3 = 1
+    x1 = DecisionVariable.binary(1)
+    x2 = DecisionVariable.binary(2)
+    x3 = DecisionVariable.binary(3)
+
+    instance = Instance.from_components(
+        decision_variables=[x1, x2, x3],
+        objective=x1 + 2 * x2 + 3 * x3,
+        constraints=[
+            x1 + x2 + x3 <= 4
+        ],  # This constraint is a meaningless constraint. However, it is set in order to specify a constraint.
+        sense=Instance.MAXIMIZE,
+    )
+
+    adapter = OMMXGurobipyAdapter(instance)
+    model = adapter.solver_input
+    model.optimize()
+    state = adapter.decode_to_state(model)
+
+    actual_entries = state.entries
+    # Should all be 1 because SOS1 constraint is not set
+    assert actual_entries[1] == pytest.approx(1)
+    assert actual_entries[2] == pytest.approx(1)
+    assert actual_entries[3] == pytest.approx(1)
+
+    assert sum(actual_entries.values()) == pytest.approx(3)
+
+
+def test_integration_with_sos1_constraint():
+    """Test problem with SOS1 constraint"""
+    # Objective function: x1 + 2 * x2 + 3 * x3
+    # Constraints:
+    #     x1 + x2 + x3 <= 4
+    #     x1, x2, x3: binary
+    # Optimal solution: x1 = 0, x2 = 0, x3 = 1
+    x1 = DecisionVariable.binary(1)
+    x2 = DecisionVariable.binary(2)
+    x3 = DecisionVariable.binary(3)
+
+    instance = Instance.from_components(
+        decision_variables=[x1, x2, x3],
+        objective=x1 + 2 * x2 + 3 * x3,  # Should select x3=1 as it has lowest cost
+        constraints=[
+            x1 + x2 + x3 <= 4
```

Comment:
```suggestion
            x1 + x2 + x3 <= 1
```

ConstraintHintと同じ制約条件を入れたほうが良いと考えます。なぜなら、OMMXの想定としてはConstraintHintはヒントであり、実態を把握するためのメタデータのようなものなので、実際に定義されている制約条件と食い違っていることは望ましくないためです。